otter/convergence/service.py

```diff
@@ -114,6 +115,10 @@ def execute_convergence(tenant_id, group_id, log,
     launch_config = manifest['launchConfiguration']
     now = yield Effect(Func(time.time))
 
+    group_status = ScalingGroupStatus.lookupByName(manifest['status'])
+    if group_status == ScalingGroupStatus.DELETING:
+        group_state.desired = 0
```

Comment:
This is a side-effect in an otherwise pure function. Also, it's not in a modify_state call, so it's not actually being persisted (unless I'm missing something), so I'm not sure why you're doing it. Doesn't the controller code already set the desired to 0 when deleting the group?